src/ui/pages/AssetInfo/AssetHistory.tsx

```diff
@@ -0,0 +1,272 @@
+  action,
+  asset,
+  address,
+  networks,
+}: {
+  action: AddressAction;
+  asset: Asset;
+  address: string;
+  networks: Networks;
+}) {
+  const { currency } = useCurrency();
+  const dialogRef = useRef<HTMLDialogElementInterface | null>(null);
+
+  const handleDialogOpen = useCallback(() => {
+    dialogRef.current?.showModal();
+  }, []);
+
+  const handleDialogDismiss = useCallback(() => {
+    dialogRef.current?.close();
+  }, []);
+
+  const { transfer, isIncoming } = useMemo(() => {
+    const incomingTransfers = action.content?.transfers?.incoming?.filter(
+      (item) => getFungibleAsset(item.asset)?.id === asset.id
+    );
+    const aggregatedIncomingTransfer: ActionTransfer | undefined =
+      incomingTransfers?.length
+        ? {
+            ...incomingTransfers[0],
+            quantity: incomingTransfers
+              .reduce((acc, item) => acc.plus(item.quantity), new BigNumber(0))
+              .toFixed(),
+          }
+        : undefined;
+    const outgoingTransfers = action.content?.transfers?.outgoing?.filter(
+      (item) => getFungibleAsset(item.asset)?.id === asset.id
+    );
+    const aggregatedOutgoingTransfer: ActionTransfer | undefined =
+      outgoingTransfers?.length
+        ? {
+            ...outgoingTransfers[0],
+            quantity: outgoingTransfers
+              .reduce((acc, item) => acc.plus(item.quantity), new BigNumber(0))
+              .toFixed(),
+          }
+        : undefined;
+    return {
+      transfer: aggregatedIncomingTransfer || aggregatedOutgoingTransfer,
+      isIncoming: Boolean(aggregatedIncomingTransfer),
+    };
+  }, [action, asset.id]);
+
+  const fungible = getFungibleAsset(transfer?.asset);
+
+  if (!fungible || !transfer) {
+    return null;
+  }
+
+  const actionType =
+    action.type.value === 'trade'
+      ? isIncoming
+        ? 'Buy'
+        : 'Sell'
+      : action.type.display_value;
+
+  const formattedPrice = transfer.price
+    ? formatPriceValue(transfer.price, 'en', currency)
+    : noValueDash;
+
+  const normalizedQuantity = baseToCommon(
+    transfer.quantity,
+    getDecimals({
+      asset: fungible,
+      chain: createChain(action.transaction.chain),
+    })
+  );
+
+  const normalizedAmountAction =
+    transfer.price === null || transfer.price === undefined
+      ? null
+      : normalizedQuantity.multipliedBy(transfer.price);
+
+  const actionTitle = `${actionType} at ${formattedPrice}`;
+  const actionDatetime = dateFormatter.format(new Date(action.datetime));
+  const actionBalance = `${isIncoming ? '+' : minus}${formatTokenValue(
+    normalizedQuantity,
+    asset.symbol,
+    { notation: normalizedQuantity.gte(100000) ? 'compact' : undefined }
+  )}`;
+  const actionValue = normalizedAmountAction
+    ? formatCurrencyValue(normalizedAmountAction, 'en', currency)
+    : null;
+
+  return (
+    <>
+      <KeyboardShortcut
+        combination="backspace"
+        onKeyDown={handleDialogDismiss}
+      />
```

Comment:
This conflicts with the Navigation Bar's backspace handler. But do we need it here? I think escape is enough